xcode/Subconscious/Shared/UIKit/BlockEditor/BlockEditorViewController.swift

```diff
@@ -0,0 +1,1211 @@
+                            at: [indexPathUp]
+                        )
+                        self.collectionView.deleteItems(
+                            at: [
+                                IndexPath(
+                                    row: indexDown,
+                                    section: BlockEditor.Section.blocks.rawValue
+                                )
+                            ]
+                        )
+                    },
+                    completion: { isComplete in
+                        self.store.send(.renderFocus(id: blockUp.id))
+                    }
+                )
+            }
+        }
+
+        return Update(state: model, render: render)
+    }
+    
+    func focus(
+        state: Model,
+        id: UUID
+    ) -> Update {
+        var model = state
+        model.blocks = state.blocks.map({ block in
+            block.setEditing(block.id == id) ?? block
+        })
+        return Update(state: model)
+    }
+    
+    func renderFocus(
+        state: Model,
+        id: UUID
+    ) -> Update {
+        guard let index = state.blocks.firstIndex(whereID: id) else {
+            Self.logger.log("block#\(id) no block found with ID. Doing nothing.")
+            return Update(state: state)
+        }
+        
+        let indexPath = IndexPath(
+            row: index,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+        
+        var model = state
+        model.blocks = state.blocks.map({ block in
+            block.setEditing(block.id == id) ?? block
+        })
+
+        let render = {
+            self.collectionView.reconfigureItems(at: [indexPath])
+        }
+        
+        return Update(state: model, render: render)
+    }
+    
+    func blur(
+        state: Model,
+        id: UUID
+    ) -> Update {
+        var model = state
+        model.blocks = state.blocks.map({ block in
+            if block.id == id {
+                return block.setEditing(false) ?? block
+            }
+            return block
+        })
+        return Update(state: model)
+    }
+    
+    func renderBlur(
+        state: Model,
+        id: UUID
+    ) -> Update {
+        guard let index = state.blocks.firstIndex(whereID: id) else {
+            Self.logger.log("block#\(id) no block found with ID. Doing nothing.")
+            return Update(state: state)
+        }
+        
+        let indexPath = IndexPath(
+            row: index,
+            section: BlockEditor.Section.blocks.rawValue
+        )
+        
+        var model = state
+        model.blocks = state.blocks.map({ block in
+            if block.id == id {
+                return block.setEditing(false) ?? block
+            }
+            return block
+        })
+        
+        let render = {
+            self.collectionView.reconfigureItems(at: [indexPath])
+        }
+        
+        return Update(state: model, render: render)
+    }
+    
+    func longPress(
```

Comment:
Just an FYI, I came across this the other day which appears to be (a) blessed way of doing multi-select gestures in collection views: https://developer.apple.com/documentation/uikit/uitableviewdelegate/selecting_multiple_items_with_a_two-finger_pan_gesture

As a user of iOS I never knew this existed but maybe we should also support it (in due time).